modules/dists/SparseBlockDist.chpl

```diff
@@ -117,6 +118,64 @@ class SparseBlockDom: BaseSparseDom {
     dsiAdd((ind,));
   }
 
+  proc dsiBulkAdd(inds: [] index(rank, idxType), isSorted=false, isUnique=false,
+      preserveInds=true){
+  
+    if !isSorted && preserveInds {
+      var _inds = inds;
+      bulkAdd_help(_inds, isSorted, isUnique); 
+    }
+    else {
+      bulkAdd_help(inds, isSorted, isUnique);
+    }
+  }
+
+  proc bulkAdd_help(inds: [] index(rank,idxType), isSorted=false, isUnique=false) {
+
+    if !isSorted then
+      QuickSort(inds);
```

Comment:
I'm a little confused by this code. Wouldn't inds be sorted by destination locale after this point? If it's block distributed? Or am I missing something?